src/app/core/teams/teams.service.spec.ts

```diff
@@ -617,56 +606,54 @@ describe('Team Service:', () => {
 			await teamsService.updateMemberRole(
 				user.explicit,
 				team.teamWithNoExternalTeam,
-				'admin'
+				TeamRoles.Admin
 			);
 
 			const uResult = await User.findById(user.explicit._id);
 			const userTeam = uResult.teams.find((t) =>
 				t._id.equals(team.teamWithNoExternalTeam._id)
 			);
 			should.exist(userTeam);
-			userTeam.role.should.equal('admin');
+			userTeam.role.should.equal(TeamRoles.Admin);
 		});
 
 		it('downgrade admin role; succeed if team has other admins', async () => {
 			await teamsService.addMemberToTeam(
 				user.user2,
 				team.teamWithNoExternalTeam,
-				'admin'
+				TeamRoles.Admin
 			);
 
 			await teamsService
-				.updateMemberRole(user.user3, team.teamWithNoExternalTeam, 'member')
+				.updateMemberRole(
+					user.user3,
+					team.teamWithNoExternalTeam,
+					TeamRoles.Member
+				)
 				.should.be.fulfilled();
 		});
 
 		it('downgrade admin role; reject if team has no other admins', async () => {
 			await teamsService
-				.updateMemberRole(user.user3, team.teamWithNoExternalTeam, 'member')
+				.updateMemberRole(
+					user.user3,
+					team.teamWithNoExternalTeam,
+					TeamRoles.Member
+				)
 				.should.be.rejectedWith({
 					status: 400,
 					type: 'bad-request',
 					message: 'Team must have at least one admin'
 				});
 		});
-
-		it('reject for invalid team role', async () => {
-			await teamsService
-				.updateMemberRole(user.user1, team.teamWithNoExternalTeam, 'fake-role')
```

Comment:
Typescript will help guard against invalid team roles, but we could theoretically receive a call that has an invalid role. Should we modify this and add the test back in?

We see a similar case here, where typescript would prevent us from passing in an invalid value, but a consumer hitting the API could send in something: https://github.com/bluehalo/node-rest-starter/pull/235/files#diff-e72196e605143b8add65be6e870db8b8d063c60afef36a9ea80ac0a63520f904L312

Reply:
> Typescript will help guard against invalid team roles, but we could theoretically receive a call that has an invalid role. Should we modify this and add the test back in?

For `updateMemberRole`, the best option would probably be to validate the incoming request to verify that a valid role is specified.  As similar approach should probably be done for `addMember` and `addMembers`.  This could be tackled as part of this if you want, or it can be deferred until later.

> We see a similar case here, where typescript would prevent us from passing in an invalid value, but a consumer hitting the API could send in something: https://github.com/bluehalo/node-rest-starter/pull/235/files#diff-e72196e605143b8add65be6e870db8b8d063c60afef36a9ea80ac0a63520f904L312

For `meetsRoleRequirement`, I think it is fine to just remove the test, since this method is always called with a static value.